src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfo.cs

```diff
@@ -785,6 +835,11 @@ internal void CacheMember(JsonPropertyInfo jsonPropertyInfo, JsonPropertyDiction
 
             if (jsonPropertyInfo.IsExtensionData)
             {
+                if (EffectiveUnmappedMemberHandling is JsonUnmappedMemberHandling.Disallow)
+                {
+                    ThrowHelper.ThrowInvalidOperationException_ExtensionDataConflictsWithUnmappedMemberHandling(Type, jsonPropertyInfo);
+                }
```

Comment:
So if a user wants to disallow unmapped members for a non-owned type-with-ext-data, they'll have to remove the ext property using a contract resolver?

Reply:
Or just remove the attribute if they can -- the two settings are mutually incompatible so only one can be set per type.